libs/ml/include/ml/graph.hpp

```diff
@@ -282,5 +301,65 @@ void Graph<ArrayType>::LoadStateDict(struct fetch::ml::StateDict<ArrayType> cons
   }
 }
 
+/**
+ * Assigns all trainable weights parameters to vector of ArrayType for exporting and serialising
+ * @return ret is vector containing values for all weights
+ */
+template <typename ArrayType>
+std::vector<ArrayType> Graph<ArrayType>::GetWeights() const
+{
+  std::vector<ArrayType> ret;
```

Comment:
point of clarification - GetWeights is not used repeatedly in a training loop (i.e. for the optimiser) correct?
because this implementation is expensive of course

Reply:
No, it is here just for test purposes, but can be used for something else in the future.